src/nanoemoji/colors.py

```diff
@@ -185,20 +186,54 @@ def color_name(rgb) -> Optional[str]:
     return None
 
 
-class Color(collections.namedtuple("Color", "red green blue alpha")):
-    _CURRENT_COLOR = (-1, -1, -1)
+@dataclasses.dataclass(frozen=True, order=True)
+class Color(Sequence):
+    red: int
+    green: int
+    blue: int
+    alpha: float = 1.0
+    index: Optional[int] = None
```

Comment:
suggest palette_index; just index isn't very obvious.

Needs a class comment to explain what it means to set or unset palette_index. For example, if I set a red value and a palette_index ... what does that mean?

Reply:
it means var(--color{self.index}, red}, like in OT-SVG color variables. Red is the default or fallback value for an implementation that does not support fetching the corresponding CPAL palette entry at given index.
I didn't use palette_index because I thought that might get confused for the index into the array of palettes, but it's actually the palette _entry_ index (which is too verbose but maybe it's clearer). Perhaps I'll change it as you suggested or add a comment

Reply:
I'd vote for long name over one that requires a comment

Reply:
palette_index() is already the name of the method that looks up a color in a list of Colors and returns the index (or 0xFFFF for currentColor). I'll rename the latter method `index_from(self, palette: Sequence["Color"]) -> int` so that I can rename the field `palette_index` as you suggested.